arcgis-ios-sdk-samples/Display information/Set up location-driven Geotriggers/SetUpLocationDrivenGeotriggersViewController.swift

```diff
@@ -0,0 +1,268 @@
+        }
+        
+        // Update status labels.
+        updateStatusLabels(featureName: featureName, notificationType: fenceNotificationInfo.fenceNotificationType)
+    }
+    
+    // MARK: UI
+    
+    func setupMapView() {
+        // Load a map with predefined tile basemap, feature styles, and labels.
+        let map = AGSMap(item: AGSPortalItem(portal: .arcGISOnline(withLoginRequired: false), itemID: "6ab0e91dc39e478cae4f408e1a36a308"))
+        map.load { [weak self] _ in
+            guard let self = self else { return }
+            
+            // Set up location display with a simulated location data source.
+            let locationDataSource = self.makeDataSource(polylineJSONString: self.walkingTourPolylineJSON)
+            self.simulatedLocationDataSource = locationDataSource
+            self.setupLocationDisplay(locationDataSource: locationDataSource)
+            
+            // Get the service feature tables from the map's operational layers.
+            if let operationalLayers = map.operationalLayers as? [AGSFeatureLayer],
+               let gardenSectionsLayer = operationalLayers.first(where: { $0.item?.itemID == "1ba816341ea04243832136379b8951d9" }),
+               let gardenPOIsLayer = operationalLayers.first(where: { $0.item?.itemID == "7c6280c290c34ae8aeb6b5c4ec841167" }),
+               let gardenSections = gardenSectionsLayer.featureTable as? AGSServiceFeatureTable,
+               let gardenPOIs = gardenPOIsLayer.featureTable as? AGSServiceFeatureTable {
+                // Create geotriggers for each of the service feature tables.
+                let geotriggerFeed = AGSLocationGeotriggerFeed(locationDataSource: locationDataSource)
+                self.createGeotriggerMonitor(feed: geotriggerFeed, featureTable: gardenSections, bufferDistance: 0.0, fenceGeotriggerName: self.sectionFenceGeotriggerName)
+                self.createGeotriggerMonitor(feed: geotriggerFeed, featureTable: gardenPOIs, bufferDistance: 10.0, fenceGeotriggerName: self.poiFenceGeotriggerName)
+            }
+        }
+        mapView.map = map
+    }
+    
+    func setupLocationDisplay(locationDataSource: AGSLocationDataSource) {
+        mapView.locationDisplay.dataSource = locationDataSource
+        mapView.locationDisplay.autoPanMode = .recenter
+        mapView.locationDisplay.initialZoomScale = 1000
+        mapView.locationDisplay.start()
+    }
+    
+    func updateStatusLabels(featureName: String, notificationType: AGSFenceNotificationType) {
+        let typeString = notificationType == .entered ? "Entered" : "Exited"
+        let fenceGeotriggerText = String(format: "%@ the geofence of %@", typeString, featureName)
+        let nearbyFeaturesText: String
+        if nearbyFeatures.keys.isEmpty {
+            nearbyFeaturesText = "No nearby features."
+        } else {
+            nearbyFeaturesText = String(format: "Nearby: %@", ListFormatter.localizedString(byJoining: Array(nearbyFeatures.keys)))
```

Comment:
Thanks for pointing this out. Initially I was thinking to preserve the insertion order of the POI features, so that the most recently entered geofence is always displayed in the first or the last popup. 
But later I realized Swift's Dictionary is unordered, in contrast to Python's dict which [preserves the insertion order](https://mail.python.org/pipermail/python-dev/2017-December/151283.html).
So I'll apply your suggestion here.